ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1180 @@
+         *          Show full path.
+         */
+        showFullPath: false,
+
+        /**
+         * CSS styles for all breadcrumbs.
+         *
+         * In styled mode, the breadcrumbs buttons are styled by the
+         * `.highcharts-range-selector-buttons .highcharts-button` rule with its
+         * different states.
+         *  @type {Highcharts.SVGAttributes}
+         *  @since     next
+         */
+        style: {
+            'fill': '#ffffff',
+            'position': 'absolute'
+        },
+
+        /**
+         * Whether to use HTML to render the breadcrumbs items texts.
+         *
+         * @type      {boolean}
+         * @since     next
+         */
+        useHTML: false,
+
+        /**
+         * The zIndex of the group.
+         *
+         * @type      {boolean}
+         * @since     next
+         */
+        zIndex: 7
+    };
+
+    /* *
+     *
+     * Properties
+     *
+     * */
+
+    public breadcrumbsGroup: SVGElement = void 0 as any;
+    public breadcrumbsList: Array<Array<any>> = [];
+    public chart: Chart;
+    public isDirty: boolean = true;
+    public level: number = -1;
+    public options: Breadcrumbs.BreadcrumbsOptions = void 0 as any;
+    public series?: TreemapSeries = void 0 as any;
+
+    public constructor(chart: Chart, userOptions?: Partial<Breadcrumbs.BreadcrumbsOptions>) {
+        const chartOptions = merge(
+            chart.options.drilldown &&
+                chart.options.drilldown.drillUpButton,
+            Breadcrumbs.defaultBreadcrumbsOptions,
+            chart.userOptions.drilldown &&
+            chart.userOptions.drilldown.drillUpButton,
+            userOptions
+        );
+
+        this.chart = chart;
+        this.options = chartOptions || {};
+    }
+
+    /**
+     * Calcule level on which chart currently is.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#calculateLevel
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public calculateLevel(this: Breadcrumbs): void {
+        const chart = this.chart,
+            drilldownLevels = chart.drilldownLevels;
+
+        // Calculate on which level we are now.
+        this.level = drilldownLevels && drilldownLevels[drilldownLevels.length - 1] &&
+        drilldownLevels[drilldownLevels.length - 1].levelNumber + 1 || 0;
+    }
+
+    /**
+     * This method creates an array of arrays containing a level number
+     * with the corresponding series/point.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#createList
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @return {any}
+     *        Formatted text.
+     */
+    public createList(this: Breadcrumbs): any {
+        const breadcrumbsList: Array<Array<number|null|Point|SeriesType|SVGAElement>> = this.breadcrumbsList || [],
```

Comment:
Q: Why we need so many different types in this array of arrays? Point/Series probably comes from Treemap vs regular drilldown, but how about the other types?

Reply:
Improved with new code version